src/modules/service-catalog/useServiceCatalogItems.tsx

```diff
@@ -0,0 +1,76 @@
+import { useCallback, useEffect, useState } from "react";
+import type { Meta } from "./data-types/Meta";
+import type { ServiceCatalogItem } from "./data-types/ServiceCatalogItem";
+
+const PAGE_SIZE = 16;
+
+export function useServiceCatalogItems(
+  searchInputValue: string,
+  currentCursor: string | null
+): {
+  serviceCatalogItems: ServiceCatalogItem[];
+  meta: Meta | null;
+  count: number;
+  isLoading: boolean;
+  errorFetchingItems: unknown;
+  fetchServiceCatalogItems: (
+    searchInputValue: string,
+    currentCursor: string | null
+  ) => void;
+} {
+  const [meta, setMeta] = useState<Meta | null>(null);
+  const [count, setCount] = useState(0);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<unknown>(null);
+  const [serviceCatalogItems, setServiceCatalogItems] = useState<
+    ServiceCatalogItem[]
+  >([]);
+
+  const fetchData = useCallback(
+    async (searchInputValue: string, currentCursor: string | null) => {
+      setIsLoading(true);
+
+      let requestUrl = "/api/v2/help_center/service_catalog/items/search?";
+
+      requestUrl +=
+        searchInputValue.length >= 3 ? `query=${searchInputValue}&` : "";
+
+      requestUrl += currentCursor
+        ? `page[size]=${PAGE_SIZE}&${currentCursor}`
+        : `page[size]=${PAGE_SIZE}`;
```

Comment:
Nit: I think it is more readable if we use [UrlSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) instead of string interpolation